x/merkledb/db.go

```diff
@@ -752,8 +752,8 @@ func (db *Database) commitChanges(ctx context.Context, trieToCommit *trieView) e
 	// invalidate all child views except for the view being committed
 	db.invalidateChildrenExcept(trieToCommit)
 
-	// move any child views of the committed trie onto the db
-	db.moveChildViewsToDB(trieToCommit)
+	// move any child views of the committed trie onto the db after changes have been made
+	defer db.moveChildViewsToDB(trieToCommit)
```

Comment:
Why is this in a defer now? We hold the database lock here so does it make a difference whether this is deferred?

Reply:
Yes, it was updating the children's parent before the data was in the cache, so children could get stale data.

Reply:
turns out this worked but wasn't the underlying issue.